wazimap_ng/datasets/admin/forms/dataset_admin_form.py

```diff
@@ -6,6 +6,17 @@
 class DatasetAdminForm(forms.ModelForm):
     import_dataset = forms.FileField(required=False)
 
+    def clean(self):
+        cleaned_data = super(DatasetAdminForm, self).clean()
+
+        profile = cleaned_data.get('profile', None)
+        permission_type = cleaned_data.get('permission_type', None)
+
+        if not profile:
```

Comment:
I would merge those together and make it more obvious
`if permission_type == 'private' and profile == None`